File: spotify-backend/music-service/src/main/java/com/spotify/musicservice/dto/Album.java
package com.spotify.musicservice.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Album {
    @JsonProperty("album_type")
    private String albumType;

    @JsonProperty("artists")
    private List<Artist> artists;

    @JsonProperty("external_urls")
    private ExternalUrls externalUrls;

    @JsonProperty("images")
    private List<Image> images;

    @JsonProperty("id")
    private String id;

    @JsonProperty("name")
    private String name;

    @JsonProperty("release_date")
    private String releaseDate;

    @JsonProperty("type")
    private String type;

    @JsonProperty("total_tracks")
    private int totalTracks;

    @JsonProperty("tracks")
    private TrackTracks tracks;
}

File: spotify-app-frontend/src/app/model/Album.ts
import { ExternalUrls } from "./ExternalUrls";
import { Artist } from "./Artist";
import { Image } from "./Image";

export class Album{
    constructor(
        public albumType: string,
        public artists: Artist[],
        public externalUrls: ExternalUrls,
        public images: Image[],
        public id: string,
        public name: string,
        public releaseDate: string,
        public type: string
    ) {}
}